packages/components/src/components/hds/dialog-primitive/footer.ts

```diff
@@ -15,14 +15,26 @@ export interface HdsDialogPrimitiveFooterSignature {
     default: [
       {
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        close?: (event: MouseEvent, ...args: any[]) => void;
+        close: (event: MouseEvent, ...args: any[]) => void;
```

Comment:
question: not sure I follow why we enforce the `close` function to exist in a block

Reply:
There were several type errors where HdsButtons in the modal footer complained that the close function could be undefined.

![Screenshot 2024-09-25 at 10 07 29 AM](https://github.com/user-attachments/assets/936759b4-4d36-419b-b5ae-c39c0a73576f)

Reply:
hm.. interesting how only by making this change I get the `modal-*.gts` components in HCP to pass linting 🤔

Reply:
could it be the case that we don't really need the fallback noop function?

Reply:
@shleewhite and I discussed this and making the `close` mandatory would not make for a flexible implementation, considering the `DisclosurePrimitive::Footer` may not contain actions; we'll stick with defining a `noop` as a fallback

Reply:
@alex-ju and I talked about this more, we decided that to make close always defined without a wrapping function would make onDismiss also required. This is a larger change that kind of goes against how the DialogPrimitive is supposed to be used, so we will stick with the wrapping function approach.